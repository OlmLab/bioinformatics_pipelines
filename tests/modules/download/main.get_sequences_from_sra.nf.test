nextflow_process {

    name "Test Process get_sequences_from_sra"
    script "modules/download/main.nf"
    process "get_sequences_from_sra"
    config "../../../configs/gutbot.config"

    test("Should run without failures") {

        when {
            process {
                """
                // define inputs of the process here. Example:
                input[0] = "SRR33759865"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
            assert process.out.fastq_files[0].size() ==2
        }

    }

}
