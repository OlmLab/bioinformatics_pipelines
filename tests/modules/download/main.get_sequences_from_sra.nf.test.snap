{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        "SRR33759865_1.fastq.gz:md5,4ad9b417c9207ab33c4eace774ab7a21",
                        "SRR33759865_2.fastq.gz:md5,2bb9759910feaa600f25255b9ff7ff79"
                    ]
                ],
                "1": [
                    "SRR33759865"
                ],
                "fastq_files": [
                    [
                        "SRR33759865_1.fastq.gz:md5,4ad9b417c9207ab33c4eace774ab7a21",
                        "SRR33759865_2.fastq.gz:md5,2bb9759910feaa600f25255b9ff7ff79"
                    ]
                ],
                "sra_ids": [
                    "SRR33759865"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-30T10:55:52.921908"
    }
}